lang/compiler.go

```diff
@@ -280,14 +280,20 @@ func (p *Program) Start(ctx context.Context, alloc memory.Allocator) (flux.Query
 	// This span gets closed by the query when it is done.
 	s, cctx := opentracing.StartSpanFromContext(ctx, "execute")
 	results := make(chan flux.Result)
+
+	resourceAlloc, ok := alloc.(*memory.ResourceAllocator)
+	if !ok {
+		resourceAlloc = &memory.ResourceAllocator{
+			Allocator: alloc,
+		}
+	}
+
 	q := &query{
 		ctx:     cctx,
 		results: results,
-		alloc: &memory.ResourceAllocator{
```

Comment:
This were creating a ResourceAllocator  that is wrapping a ResourceAllocator which seems redundant at best, but was causing errors due to setting multiple finalizers on the same pointer.